lib/partial/transmuxer.js

```diff
@@ -0,0 +1,375 @@
+var Stream = require('../utils/stream.js');
+var m2ts = require('../m2ts/m2ts.js');
+var codecs = require('../codecs/index.js');
+var AudioSegmentStream = require('./audio-segment-stream.js');
+var VideoSegmentStream = require('./video-segment-stream.js');
+var trackInfo = require('../mp4/track-decode-info.js');
+var isLikelyAacData = require('../aac/utils').isLikelyAacData;
+var AdtsStream = require('../codecs/adts');
+var AacStream = require('../aac/index');
+var clock = require('../utils/clock');
+
+var createPipeline = function(object) {
+  object.prototype = new Stream();
+  object.prototype.init.call(object);
+
+  return object;
+};
+
+var tsPipeline = function(options) {
+  var
+    pipeline = {
+      type: 'ts',
+      tracks: {
+        audio: null,
+        video: null
+      },
+      packet: new m2ts.TransportPacketStream(),
+      parse: new m2ts.TransportParseStream(),
+      elementary: new m2ts.ElementaryStream(),
+      videoRollover: new m2ts.TimestampRolloverStream('video'),
+      audioRollover: new m2ts.TimestampRolloverStream('audio'),
+      adts: new codecs.Adts(),
+      h264: new codecs.h264.H264Stream(),
+      captionStream: new m2ts.CaptionStream(),
+      metadataStream: new m2ts.MetadataStream(),
+      timedMetadataTimestampRolloverStream:
+        new m2ts.TimestampRolloverStream('timed-metadata')
+  };
+
+  pipeline.headOfPipeline = pipeline.packet;
+
+  // Transport Stream
+  pipeline.packet
+    .pipe(pipeline.parse)
+    .pipe(pipeline.elementary);
+
+  // H264
+  pipeline.elementary
+    .pipe(pipeline.videoRollover)
+    .pipe(pipeline.h264);
+
+  // Hook up CEA-608/708 caption stream
+  pipeline.h264
+    .pipe(pipeline.captionStream);
+
+  pipeline.elementary
+    .pipe(pipeline.timedMetadataTimestampRolloverStream)
+    .pipe(pipeline.metadataStream);
+
+  // ADTS
+  pipeline.elementary
+    .pipe(pipeline.audioRollover)
+    .pipe(pipeline.adts);
+
+  pipeline.elementary.on('data', function(data) {
+    if (data.type !== 'metadata') {
+      return;
+    }
+
+    for (var i = 0; i < data.tracks.length; i++) {
+      if (!pipeline.tracks[data.tracks[i].type]) {
+        pipeline.tracks[data.tracks[i].type] = data.tracks[i];
+      }
+    }
+
+    if (pipeline.tracks.video && !pipeline.videoSegmentStream) {
+      pipeline.videoSegmentStream = new VideoSegmentStream(pipeline.tracks.video, options);
+
+      pipeline.videoSegmentStream.on('timelineStartInfo', function(timelineStartInfo) {
+        if (pipeline.tracks.audio) {
+          pipeline.audioSegmentStream.setEarliestDts(timelineStartInfo);
```

Comment:
The non-partial transmuxer sets `timelineStartInfo` on the audio track here which you are no longer doing. 
There is the comment 
```
          // When video emits timelineStartInfo data after a flush, we forward that
          // info to the AudioSegmentStream, if it exists, because video timeline
          // data takes precedence.
```

Is this no longer the case? are `timelineStartInfo` now being kept completely separate?

Reply:
Nice catch.

In the non-partial transmuxer we only seem to use the `timelineStartInfo` in the audio track for `setEarliestDts`, where the earliest allowed dts would be the dts passed in as an argument minus the `timelineStartInfo.baseMediaDecodeTime`. Technically we can probably simplify the logic over in the non-partial transmuxer by removing that and doing the calculation before passing in the earliest dts.

In the partial transmuxer, things are a bit different, so we'll have to play with the logic a bit. `timelineStartInfo` is now triggered in the video segment stream before the first recalculation of `baseMediaDecodeTime`, whereas in the non-partial it was triggered after, thus, could be a different value. For the case of keeping original timestamps, we may be OK, but the logic may have to be changed around a bit.